docs/proposals/20250425-aro-hcp.md

```diff
@@ -0,0 +1,684 @@
+ ChannelGroup string `json:"channelGroup"`
+
+ // VersionGate requires acknowledgment when upgrading ARO-HCP y-stream versions (e.g., from 4.20 to 4.21).
+ // Allowed values are Acknowledge, WaitForAcknowledge and AlwaysAcknowledge default is WaitForAcknowledge.
+ VersionGate string `json:"versionGate"`
+
+
+ // IdentityRef is a reference to an identity to be used when reconciling the aro control plane.
+ // If no identity is specified, the default identity for this controller will be used.
+ IdentityRef *infrav1.AzureClusterIdentity `json:"identityRef,omitempty"`
+
+
+ // AdditionalTags are user-defined tags to be added to the Azure resources associated with the control plane.
+ AdditionalTags infrav1.Tags `json:"additionalTags,omitempty"`
+}
+
+
+// PlatformProfile represents the Azure platform configuration.
+type PlatformProfile struct {
+ // Location should be valid Azure location ex; centralus
+ Location string `json:"location,omitempty"`
+
+
+ // Resource group name where the Aro-hcp will be attached to it.
+ ResourceGroup string `json:"resourceGroup,omitempty"`
+
+
+ // ResourceGroup Ref name that is used to create the ResourceGroup CR. The ResourceGroupRef must be in the same namespace as the AroControlPlane and cannot be set with ResourceGroup.
+ ResourceGroupRef string `json:"resourceGroup,omitempty"`
+
+
+ // Azure subnet id
+ Subnet string `json:"subnet,omitempty"`
+
+
+ // Subnet Ref name that is used to create the VirtualNetworksSubnet CR. The SubnetRef must be in the same namespace as the AroControlPlane and cannot be set with Subnet.
+ SubnetRef string `json:"subnetRef,omitempty"`
+
+
+  // OutboundType represents a routing strategy to provide egress to the Internet. Allowed value is loadBalancer
+  OutboundType string `json:"outboundType,omitempty"`
+
+  // Azure Network Security Group ID
+  NetworkSecurityGroupID string `json:"networkSecurityGroupId,omitempty"`
+
+  // ManagedIdentities Azure managed identities for ARO HCP.
+  ManagedIdentities ManagedIdentities `json:"managedIdentities,omitempty"`
+}
+
+type ManagedIdentities struct {
+  // CreateAROHCPManagedIdentities is used to create the required ARO-HCP managed identities if not provided.
+  // It will create UserAssignedIdentity CR for each required managed identity. Default is false. 
+  CreateAROHCPManagedIdentities bool `json:"createAROHCPManagedIdentities,omitempty"`
+  
+ // ControlPlaneOperators Ref to Microsoft.ManagedIdentity/userAssignedIdentities
+ ControlPlaneOperators  *ControlPlaneOperators `json:"controlPlaneOperators,omitempty"`
+
+ // DataPlaneOperators ref to Microsoft.ManagedIdentity/userAssignedIdentities
+ DataPlaneOperators     *DataPlaneOperators `json:"dataPlaneOperators,omitempty"`
+
+ // ServiceManagedIdentity ref to Microsoft.ManagedIdentity/userAssignedIdentities
+ ServiceManagedIdentity string            `json:"serviceManagedIdentity,omitempty"`
+}
+
+type ControlPlaneOperators struct {
```

Comment:
Are we not planning to reuse the HyperShift API for this, rather than duplicating API efforts?

Reply:
no, we will re-use the Hypershift API. The naming for CRD fields is a bit different but internally with API call will set the naming as [here](https://github.com/Azure/ARO-HCP/blob/main/cluster-service/cluster-creation.md?plain=1#L174)